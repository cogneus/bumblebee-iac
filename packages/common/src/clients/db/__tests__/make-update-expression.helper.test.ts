import { makeUpdateExpression } from '../make-update-expression.helper'

describe('Make update expression tests', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('should match exporession output', () => {
    expect(
      makeUpdateExpression({
        value1: 'test value',
        value2: true,
        value3: 1,
        value4: undefined,
        value5: '',
      })
    ).toMatchSnapshot()
  })
})
