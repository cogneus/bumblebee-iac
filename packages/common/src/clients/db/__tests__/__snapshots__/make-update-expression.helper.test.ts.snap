// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make update expression tests should match exporession output 1`] = `
{
  "ExpressionAttributeNames": {
    "#value1": "value1",
    "#value2": "value2",
    "#value3": "value3",
    "#value4": "value4",
    "#value5": "value5",
  },
  "ExpressionAttributeValues": {
    ":value1": "test value",
    ":value2": true,
    ":value3": 1,
    ":value4": null,
    ":value5": "",
  },
  "UpdateExpression": "SET #value1 = :value1, #value2 = :value2, #value3 = :value3, #value4 = :value4, #value5 = :value5",
}
`;
